### PR TITLE
Add workflows

### PR DESCRIPTION
**Pull request by mcalus3**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
```
Comment (punkle) [inline .github/workflows/codeql-analysis.yml:25]: I think we can remove these comments.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['javascript']
+        # Learn more...
```
Comment (punkle) [inline .github/workflows/codeql-analysis.yml:28]: likewise

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
```
Comment (punkle) [inline .github/workflows/codeql-analysis.yml:40]: What are we testing the head of the pull request or the merge? Why do we want to test the pull request and not what will be merged?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
```
Comment (punkle) [inline .github/workflows/main.yml:11]: I think we can remove this space.

**Comment by punkle** [comment]
Should make these generic workflows and put them in the .github repo so that we can reuse them across the other plugins?

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (martina-if) [reply .github/workflows/codeql-analysis.yml:40]: yeah this is a bit weird, this is missing a couple of commits :thinking:

Comment (iain-b) [reply .github/workflows/codeql-analysis.yml:40]: So this file was auto-generated and originally added to travis CI. This approach was the recommended practice https://github.com/github/codeql-action/issues/105. It seems like it is no longer but I need to check.

Comment (iain-b) [reply .github/workflows/codeql-analysis.yml:40]: https://github.com/github/codeql-action/commit/988704e971030d5d57fbe56e952901d787f31852#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 Changed about a month ago, I can't seem to find a PR or issue with details but safe to remove I think (we can always add it back if the line numbers are wonky).

**Review by iain-b — APPROVED** [review]

**Pull request merged**